pkgs/build-support/build-fhsenv-bubblewrap/default.nix

```diff
@@ -27,12 +27,15 @@
 , ...
 } @ args:
 
+assert (pname != null || name != null);
 assert (pname != null || version != null) -> (name == null && pname != null); # You must declare either a name or pname + version (preferred).
 
 with builtins;
 let
-  pname = if args ? name && args.name != null then args.name else args.pname;
-  versionStr = lib.optionalString (version != null) ("-" + version);
+  pname = if args ? pname && args.pname != null then args.pname else lib.getName args.name;
+  versionStr = lib.optionalString (version != null || name != null) ("-" + (if version != null then version else (lib.getVersion name)));
```

Comment:
The `version` attribute of the final derivation is still `null` when only `name` is passed.
<https://github.com/NixOS/nixpkgs/blob/8fc9bb20e48ad01bd837726f48e1575aedbee9a5/pkgs/build-support/build-fhsenv-bubblewrap/default.nix#L226>

For example:
```console
$ nix eval github:nixos/nixpkgs/pull/262506/head#legacyPackages.x86_64-linux.osu-lazer-bin.pname
"osu-lazer-bin"
$ nix eval github:nixos/nixpkgs/pull/262506/head#legacyPackages.x86_64-linux.osu-lazer-bin.version
null
```

I think it is a little inconsistent.

Maybe we should do something similar to `version`.
```nix
# lib.getVersion returns "" on failure
versionFromName = let v = lib.getVersion name; in if v == "" then null else v;
version = if args ? version && args.version != null then args.version else versionFromName;
versionStr = lib.optionalString (version != null) ("-" + version)
```

Reply:
Done, please test using the following:

```nix
test = drv: { "pname" = drv.pname or "<no-pname>"; "name" = drv.name or "<no-name>"; "version" = drv.version or "<no-version>"; }
# These should succeed
test (buildFHSEnvBubblewrap { "pname" = "hello"; "version" = "1.2.3"; }) # { name = "hello-1.2.3"; pname = "hello"; version = "1.2.3"; }
test (buildFHSEnvBubblewrap { "name" = "hello-1.2.3"; }) # { name = "hello-1.2.3"; pname = "hello"; version = "1.2.3"; }
test (buildFHSEnvBubblewrap { "name" = "hello"; }) # { name = "hello"; pname = "hello"; version = null; }
test (buildFHSEnvBubblewrap { "pname" = "hello"; }) # { name = "hello"; pname = "hello"; version = null; }
```

To avoid repeating the test on presence of `args.pname`, I used an attrset to set `pname` and `version` at the same time. Please let me know whether you think that is appropriate.

BTW, I start to doubt whether it's OK for a derivation to have both `name` and `pname`. Note that is https://github.com/NixOS/nixpkgs/commit/055b69b76dc66c24d7b948b2c913eaf438194fbd#diff-61606177a03a1eb20736852c5d456ad6d4da6742367795d1d452a29a6d8605a5 that starts to preserve `pname` in `buildFHSEnvBubblewrap`; before that the result derivation contains only `name`.